feat(platform): Creating search input component

KevinOkamoto opened the pull request:
#### Please provide a link to the associated issue.
#1246

#### Please provide a brief summary of this pull request.
This is an initial PR for the SearchInputComponent for `platform` as described here:

https://github.com/SAP/fundamental-ngx/wiki/Platform:-SearchInput-Technical-Design


#### If this is a new feature, have you updated the documentation?

fkolar commented:
### search-input.component.html 

`[triggers]="[]"` - there is some default value already. Or are you trying to have no event triggers at all ?

### PlatformSearchInputComponent

  - Is there a reason why we need to explicitelly call this `PlatformSearchInputComponent`? I woudl just call it `SearchInputComponent `, we can still import both from code, platform and give it alias. The import statement can do this for us. 

  - `import { ComboboxComponent, ComboboxItem } from '@fundamental-ngx/core';` The comboboxItem is not exported from core.

  - `ngAfterViewInit`  - cannot this `this.combobox` be null and cause problem at some cases ? 

 - Just to make sure this component will never be part of the form, its always meant to be page level component, so we dont need to support forms at all. 

 - Haven't we said that we will support both list of values and observable stream? 

 - We should have changedetection as `OnPush` to be more peformance driven and to guarantee that our component will not be cause perf. problems

- Let's put a directive as first thing in the `li` tag so its recognizable easier
```
<li *ngFor="let category of categories" (click)="setCurrentCategory(category)"><span
                            fd-menu-item>{{category.label}}</span></li> 

```

KevinOkamoto commented:
> `[triggers]="[]"` - there is some default value already. Or are you trying to have no event triggers at all ?

Yes, I'm purposely trying to disable the `triggers`. By default the combobox opens the dropdown on "click" of the input box - I did not want the dropdown box to open until the user starts typing and suggestions are provided.

KevinOkamoto commented:
> ### PlatformSearchInputComponent
> * Is there a reason why we need to explicitelly call this `PlatformSearchInputComponent`? I woudl just call it `SearchInputComponent `, we can still import both from code, platform and give it alias. The import statement can do this for us.

Initially I wanted to do this so that we could import both libraries wholely without having to import individual components. However, since core has removed SearchInputComponent from their library, I can make this name change.

> * `import { ComboboxComponent, ComboboxItem } from '@fundamental-ngx/core';` The comboboxItem is not exported from core.

I added the export of this interface in this PR.

> * `ngAfterViewInit`  - cannot this `this.combobox` be null and cause problem at some cases ?

I've already added "if-guards" if this is the case.

> * Just to make sure this component will never be part of the form, its always meant to be page level component, so we dont need to support forms at all.

As far as I know, this is the case.

> * Haven't we said that we will support both list of values and observable stream?

Yes, I need to implement this.

> * We should have changedetection as `OnPush` to be more peformance driven and to guarantee that our component will not be cause perf. problems

OK

> * Let's put a directive as first thing in the `li` tag so its recognizable easier
> 
> ```
> <li *ngFor="let category of categories" (click)="setCurrentCategory(category)"><span
>                             fd-menu-item>{{category.label}}</span></li> 
> ```

OK
